### PR TITLE
Outreach filteres added and sorted

### PR DESCRIPTION
**Pull request by jtmst**
## Description
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11153

Adjusted graphql query to allow outreach filters to be added dynamically

## Testing done
Local testing with backend tugboat instance

## Screenshots
Screen Shot 2022-11-02 at 9 45 32 AM
Screen Shot 2022-11-04 at 1 14 48 PM

## Acceptance criteria
- [ ] The topic filter options are powered by field_lc_categories from the data items on the page (this is a new field and will need to be added to the graphql.)
- [ ] ~~The file type filter options are powered by field_format  (this is not a new field)~~
- [ ] Topic filter and type filter select lists are alpha sorted
- [ ] A veteran should never be able to select a topic option that has no data. (example: Disability should not appear as a filter if there are no data items of topic Disability.)
- [ ] The field field_benefits is no longer referenced on the template and no longer in the graphQL 
- [ ] Accessibility review - receives 508 approval 
- [ ] Create future issue -  The two filters are aware of each other and self restrict the other's limits.  (example: If I select format = video, I should not see types in the filter that contain no videos. This would prevent from selecting a filter combination that has no results.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/filters/liquid.js [hunk 1]
```diff
@@ -109,6 +109,30 @@ module.exports = function registerFilters() {
 
   liquid.filters.formatDate = (dt, format) => prettyTimeFormatted(dt, format);
 
+  liquid.filters.buildTopicList = topics => {
+    if (!topics) return null;
+    const topicArray = [];
+
```
Comment (apisandipas) [inline src/site/filters/liquid.js:115]: For building up an array like this, I think it's a little more idiomatic and easier to read if you use `reduce` function. I'm interested to know @ryguyk's take on that though.

**Review by apisandipas — CHANGES_REQUESTED** [review]
I had one concern/comment/question. I don't think it should be a blocker, but perhaps check with @ryguyk for his opinion.

**Review by apisandipas — APPROVED** [review]
LGTM, refactor looks good .

**Comment by jilladams** [comment]
One flag here: 
https://prod.cms.va.gov/admin/content?title=PACT&type=outreach_asset&moderation_state=All&owner=All -- in Prod we now have a batch of Publications using the PACT act taxonomy, as of EOD yesterday 11/8

Verified that those nodes contain that data on Staging, e.g. https://staging.cms.va.gov/node/50916/edit

But on Staging (where this fix is now deployed), filtering by PACT act doesn't show any publications: https://staging.va.gov/outreach-and-events/outreach-materials/

**Comment by jilladams** [comment]
(Will take a look once it's on prod, just in case this is some Staging data weirdness.)

**Pull request merged**